siliconcompiler/scheduler/scheduler.py

```
@@ -311,15 +311,20 @@ def run(self) -> None:
             if self.__check_collect_files():
                 collect(self.project)
 
-            self.run_core()
 
-            # Store run in history
-            self.__project._record_history()
+            try:
+                self.run_core()
+            except SCRuntimeError as e:
+                raise e
```

Ruff (0.14.7)

Remove exception handler; error is immediately re-raised

(TRY203)